tokio/src/runtime/task/mod.rs

```diff
@@ -201,10 +201,21 @@ use std::{fmt, mem};
 /// [unstable]: crate#unstable-features
 #[cfg_attr(docsrs, doc(cfg(all(feature = "rt", tokio_unstable))))]
 #[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
-// TODO(eliza): there's almost certainly no reason not to make this `Copy` as well...
-#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
 pub struct Id(u64);
 
+/// Returns the `Id` of the task.
+///
+/// # Panics
+///
+/// This function panics if called from outside a task.
+///
+#[allow(unreachable_pub)]
+pub fn id() -> Id {
```

Comment:
I would prefer to use a different name here. Probably something that contains "current".

Reply:
Something with "current" makes sense in within the context of "Context", because different tasks come and go, but in the context of a "task", I don't think it makes sense -- a task has a single id during its lifetime, so I think task::id() is better.

Reply:
@hawkw thanks for the review!

I like ```task::id()``` better because it is what the name says. When I see that call, it is clear that the it returns the task's id. When I see ```Id::current()```, it is confusing because what would a non-current id be? If I call it again later, will the id be different? But I won't fight for it -- if you folks think ```Id::current()``` is better I can change it to that.